File: backend/src/main/java/com/phishingdefense/backend/entity/Stage.java
package com.phishingdefense.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * PRD상 "시나리오(scenario)"에 해당하는 테이블을 매핑한다.
 * 게임 내 용어로는 챕터 하위의 "스테이지"로 취급한다.
 */
@Entity
@Table(name = "scenarios")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Stage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "scenario_id")
    private Long stageId;

    @Column(name = "chapter_id", nullable = false)
    private Integer chapterId;

    @Column(name = "title")
    private String title;

    @Column(name = "initial_message")
    private String initialMessage;

    @Column(name = "phishing_type")
    private String phishingType;

    @Column(name = "is_phishing")
    private Boolean phishing;

    @Column(name = "required_evidence", columnDefinition = "JSON")
    private String requiredEvidence;

    @Column(name = "difficulty")
    private Integer difficulty;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: mobile/lib/models/game/stage.dart
class Stage {
  const Stage({
    required this.stageId,
    required this.chapterId,
    required this.title,
    required this.initialMessage,
    required this.phishingType,
    required this.difficulty,
    required this.completed,
  });

  factory Stage.fromJson(Map<String, dynamic> json) {
    return Stage(
      stageId: json['stageId'] as int,
      chapterId: json['chapterId'] as int,
      title: json['title'] as String,
      initialMessage: json['initialMessage'] as String,
      phishingType: json['phishingType'] as String? ?? 'unknown',
      difficulty: json['difficulty'] as int? ?? 1,
      completed: json['completed'] as bool? ?? false,
    );
  }

  final int stageId;
  final int chapterId;
  final String title;
  final String initialMessage;
  final String phishingType;
  final int difficulty;
  final bool completed;

  String get difficultyLabel {
    switch (difficulty) {
      case 1:
        return '쉬움';
      case 2:
        return '보통';
      default:
        return '어려움';
    }
  }
}
